ingestion/src/metadata/parsers/avro_parser.py

```diff
@@ -27,6 +27,7 @@
 logger = ingestion_logger()
 
 RECORD_DATATYPE_NAME = "RECORD"
+PARSED_ALREADY = {}
```

Comment:
since we are using a global variable, add a test that runs this twice to validate there are no side effects.

Reply:
I'm doing a `PARSED_ALREADY.clear()` before we return the parsed schema to take care of it.
In the test this scenario has been already taken care of as we have mutiple sample schemas having the same name and type in it.

Reply:
why is it global?

Reply:
To parse the avro schema we are having an recursive approach for its fields.
To check if a schema is parsed already its declared out of function.

Reply:
If its a recursive function the correct way is to have the state in the function signature. Something like:

```python
def parse_avro_schema(
    schema: str, cls: Type[BaseModel] = FieldModel, already_parsed
) -> Optional[List[Union[FieldModel, Column]]]:
```

Then start with the initial state:

```python
parse_avro_schema(schema,cls,[])
```

Reply:
yes right. i'll put it in a function signature

Reply:
you can set the the `already_parsed` as optional and initialize it to an empty list if its `None`. That way the first call will initialize the recursion stack.